app/code/Magento/CustomerGraphQl/Model/Resolver/CreateCustomer.php

```diff
@@ -0,0 +1,130 @@
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Newsletter\Model\SubscriberFactory;
+use Magento\Store\Model\StoreManagerInterface;
+use Magento\Framework\Api\DataObjectHelper;
+
+/**
+ * Create customer data resolver
+ */
+class CreateCustomer implements ResolverInterface
+{
+    /**
+     * @var CustomerDataProvider
+     */
+    private $customerDataProvider;
+
+    /**
+     * @var AccountManagementInterface
+     */
+    private $accountManagement;
+
+    /**
+     * @var CustomerInterfaceFactory
+     */
+    private $customerFactory;
+
+    /**
+     * @var DataObjectHelper
+     */
+    private $dataObjectHelper;
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+    /**
+     * @var SubscriberFactory
+     */
+    private $subscriberFactory;
+
+    /**
+     * @param DataObjectHelper $dataObjectHelper
+     * @param CustomerInterfaceFactory $customerFactory
+     * @param AccountManagementInterface $accountManagement
+     * @param StoreManagerInterface $storeManager
+     * @param SubscriberFactory $subscriberFactory
+     * @param CustomerDataProvider $customerDataProvider
+     */
+    public function __construct(
+        DataObjectHelper $dataObjectHelper,
+        CustomerInterfaceFactory $customerFactory,
+        AccountManagementInterface $accountManagement,
+        StoreManagerInterface $storeManager,
+        SubscriberFactory $subscriberFactory,
+        CustomerDataProvider $customerDataProvider
+    ) {
+        $this->customerDataProvider = $customerDataProvider;
+        $this->accountManagement = $accountManagement;
+        $this->customerFactory = $customerFactory;
+        $this->dataObjectHelper = $dataObjectHelper;
+        $this->storeManager = $storeManager;
+        $this->subscriberFactory = $subscriberFactory;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        if (!isset($args['input']) || !is_array($args['input']) || empty($args['input'])) {
+            throw new GraphQlInputException(__('"input" value should be specified'));
+        }
+        try {
+            $customer = $this->createUserAccount($args);
+            $this->setUpUserContext($context, $customer);
+            if (array_key_exists('is_subscribed', $args['input'])) {
+                if ($args['input']['is_subscribed']) {
+                    $this->subscriberFactory->create()->subscribeCustomerById($customer->getId());
+                }
+            }
+            $data = $this->customerDataProvider->getCustomerById((int)$customer->getId());
+        } catch (LocalizedException $e) {
+            throw new GraphQlInputException(__($e->getMessage()));
+        }
+        return ['customer' => $data];
+    }
+
+    private function createUserAccount($args)
```

Comment:
Pls, add dockblcoks and type hinting to all of methods (including return values)